classes/class3/README.md

```diff
@@ -1 +1,364 @@
+
+First, were going to install 3 more modules that are used in pretty much every Express app: `npm install --save body-parser cookie-parser morgan`.
+Next we'll use them in `app.js` by doing the following:
+Add these lines to the top of the file, next to the rest of the requires:
+```javascript
+var path = require('path');
+var logger = require('morgan');
+var cookieParser = require('cookie-parser');
+var bodyParser = require('body-parser');
+```
+We add `path` here which we did not need to install because it is included within node. It gives us some useful directory path functionality you'll see in a second.
+Add these lines of configuration below the line `var app = express();`:
+```javascript
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+```
+This sets up some basic command line logging functionality using the [Morgan](https://github.com/expressjs/morgan) module, request body parsing, cookie parsing, and establishes the `public` directory as a static folder. This means that going to `http://localhost:3000/images/foo.jpg` will load the file `./public/images/foo.jpg` without any additional work from us. The same will work for javascript files, stylesheets, static html files, or anything else you may want to serve directly.
+
+If you run the app again (`node app`) upon navigating to http://localhost:3000 you will notice a new line in your console:
+```
+GET / 304 6.892 ms - -
+```
+This is Morgan logging the request. This will really come in handy while trying to debug in the future.
+
+Now go ahead and copy the image `cat.jpg` from the `expressintro/public/images` folder in this directory and place it in your new images folder. Then navigate to http://localhost:3000/images/cat.jpg. That is your static folder doing work!
+
+## Templating
+
+Now I imagine you may be asking yourself, *"This is a web-dev course, right? Where's the HTML?"* Well its coming, but doing it smart is not as simple as putting some `<p>` tags in a file and calling it a day. We will be using a technique called templating. Templating allows us to put some logic behind what is otherwise entirely a layout language. This will greatly reduce the overall quantity of HTML you write, thereby reducing errors, and making it much easier to make a change to an element that appears on many pages in your app. Most importantly however, it allows us dynamically customize the contents of our HTML. Think about Facebook profiles. When you load yours, you are looking at a static page of HTML. But Facebook's servers aren't full of files called `ben-kahle-profile.html`, and `evan-simpson-profile.html` that wouldn't exactly scale very well. They have a single profile template into which they inject your personal profile information, render it as HTML and then send it to your browser. Let's learn the basics!
```

Comment:
us dynamically -> us to dynamically
load yours -> load your profile
...`.html` that wouldn't -> ...`.html`; that wouldn't